JetNews/app/src/main/java/com/example/jetnews/ui/interests/InterestsScreen.kt

```diff
@@ -162,30 +158,72 @@ fun InterestsScreen(
         }
     ) { innerPadding ->
         val screenModifier = Modifier.padding(innerPadding)
-        TabContent(windowSize, currentSection, onTabChange, tabContent, screenModifier)
+        InterestScreenContent(
+            isTabRowExpanded, currentSection, onTabChange, tabContent, screenModifier
+        )
+    }
+}
+
+@Composable
+fun rememberTabContent(interestsViewModel: InterestsViewModel): List<TabContent> {
+    // UiState of the InterestsScreen
+    val uiState by interestsViewModel.uiState.collectAsState()
+
+    // Describe the screen sections here since each section needs 2 states and 1 event.
+    // Pass them to the stateless InterestsScreen using a tabContent.
+    val topicsSection = TabContent(Sections.Topics) {
+        val selectedTopics by interestsViewModel.selectedTopics.collectAsState()
+        TabContent(
+            sections = uiState.topics,
+            selectedTopics = selectedTopics,
+            onTopicSelect = { interestsViewModel.toggleTopicSelection(it) }
+        )
+    }
+
+    val peopleSection = TabContent(Sections.People) {
+        val selectedPeople by interestsViewModel.selectedPeople.collectAsState()
+        val section = listOf(InterestSection("", uiState.people))
```

Comment:
Not a big fan of using the empty string here as a placeholder... it also looks like selecting is now broken for `People` and `Publication` as a related issue.

Reply:
Yeah, didn't like it either. Reverting back but reusing as much as possible. Thanks!